### DISCUSSION TITLE
unit test with reactjs and vitest

### DISCUSSION THREAD

**Discussion by malomiquel** [clientIO/joint#2198] [Q&A]
### Introduction

Hello everyone,
I need some help on units tests using vitest. I don't understand why I have this error:

![image](https://github.com/clientIO/joint/assets/83480447/799c5185-a6a9-40dc-80f2-c6b178c0b87f)

Here is my tests:

![code](https://github.com/clientIO/joint/assets/83480447/0d6c89e6-3129-4008-8f26-7490ef7946d1)

My code:

![code](https://github.com/clientIO/joint/assets/83480447/b1856ead-b9d6-4e4e-942e-1ec1588bb968)

And my setupTests:

![code](https://github.com/clientIO/joint/assets/83480447/ed1340fb-57a2-4b8d-9786-827ff2b932ba)

Thank for your help

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by jamesgeorgewilliams** [comment]
It would be great if you can provide a repository with a minimal example that reproduces the error, and then we will take a look. Thanks

**Reply by malomiquel** [reply]
I'm only experiencing this error with these particular tests. I can't provide any more information than what I've already given. I used Jest before switching to Vitest and didn't encounter any errors, which is why I'm finding this situation difficult to understand.

Here is my vite.config file.
![code](https://github.com/clientIO/joint/assets/83480447/d1dd0905-779b-4006-a60e-7d68be5a500c)

**Reply by kumilingus** [reply]
You can provide us with a repository that we can install, run, observe and fix. Thank you.

**Reply by malomiquel** [reply]
Yes of course here his a project with same configurations tests:
https://github.com/malomiquel/jointjs-tests
You can run tests

**Reply by jamesgeorgewilliams** [reply]
@malomiquel Just out of curiosity I tried these same tests in Jest, and I get the same error. Did you have these exact tests passing successfully in Jest before, or I misunderstood?

**Reply by malomiquel** [reply]
Yes, it's probably not the same tests as with jest. I can't remember which tests I did to get them to pass with Jest.
